src/components/Products/ProductDetails.js

```diff
@@ -0,0 +1,94 @@
+import { useParams } from "react-router";
+import { useOffers } from "../../context/offers-context";
+import { useProduct } from "../../context/product-context";
+import { GiRoundStar } from "react-icons/gi";
+import "../../styles.css";
+import { AddToCart } from "../Cart/AddToCart";
+import { AddToWishlist } from "../Wishlist/AddToWishlist";
+
+export const ProductDetails = () => {
+  const { offersList } = useOffers();
+  const { productList } = useProduct();
+  const { productId } = useParams();
+  const product = productList.find((item) => item._id === productId);
+  const offerId = product.offers[0];
+  const offer = offersList.find((item) => item._id === offerId);
+  console.log(offer);
```

Comment:
Remove all the consoles. There are many.

Reply:
yes!